stackrox/stackrox
# ROX-20352: Reconcile ocp resources

## Description

Sensor was not tracking `compliance operator` resources. This PR aims to introduce a new store to track these resources and upon connection with Central, Sensor will do the reconciliation of these resources.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

   * In an OCP cluster, install the [compliance-operator](https://docs.openshift.com/acs/operating/manage-compliance-operator/compliance-operator-rhacs.html)
   * Deploy ACS
   * Perform an scan with ACS: Compliance -> Scan environment
   * Check central-db and see that the following tables are populated:
     * compliance_operator_check_results
     * compliance_operator_profiles
     * compliance_operator_rules
     * compliance_operator_scan_setting_bindings
     * compliance_operator_scans
   * Scale down sensor
   * Delete the `compliance-operator`
   * Scale up sensor
   * The tables listed above should be empty

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.
